src/dra/5D874.c

```diff
@@ -2143,9 +2143,171 @@ void func_80123A60(Entity* entity) {
 
 INCLUDE_ASM("asm/dra/nonmatchings/5D874", func_80123B40);
 
-INCLUDE_ASM("asm/dra/nonmatchings/5D874", func_80123F78);
+void func_80123F78(Entity* entity) {
+    if (D_800973FC == 0) {
+        D_80097F3F = 0;
+        func_80106590(entity);
+        return;
+    }
 
-INCLUDE_ASM("asm/dra/nonmatchings/5D874", func_80124164);
+    switch (entity->step) {
+    case 0:
+        entity->unk34 = 0x0C030000;
+        if (PLAYER.animationSet != 1) {
+            func_80106590(entity);
+            break;
+        }
+        entity->animationSet = 1;
+        entity->animationFrame = PLAYER.animationFrame;
+        entity->unk5A = 0xD;
+        entity->unk19 = PLAYER.unk19;
+        entity->blendMode = 0x30;
+        entity->palette = 0x815F;
+        entity->zPriority = PLAYER.zPriority - 2;
+        entity->facing = PLAYER.facing;
+        do { // TODO: !FAKE
+        } while (0);
+        D_80097F4A = 0x80;
+        D_80097F46 = 0x80;
+        D_80097F42 = 0x80;
+        D_80097F3E = 0x80;
+        D_80097F49 = 0x80;
+        D_80097F45 = 0x80;
+        D_80097F41 = 0x80;
+        D_80097F3D = 0x80;
+        D_80097F48 = 0x80;
+        D_80097F44 = 0x80;
+        D_80097F40 = 0x80;
+        D_80097F3C = 0x80;
+        D_80097F3F = 1;
+        do { // TODO: !FAKE
+        } while (0);
+        entity->step++;
+        break;
+
+    case 1:
+        D_80097F4A += 255;
+        if (D_80097F4A < 4) {
+            D_80097F3F = 0;
+            func_80106590(entity);
+            break;
+        }
+        D_80097F46 = D_80097F4A;
+        D_80097F42 = D_80097F4A;
+        D_80097F3E = D_80097F4A;
+        D_80097F49 = D_80097F4A;
+        D_80097F45 = D_80097F4A;
+        D_80097F41 = D_80097F4A;
+        D_80097F3D = D_80097F4A;
+        D_80097F48 = D_80097F4A;
+        D_80097F44 = D_80097F4A;
+        D_80097F40 = D_80097F4A;
+        D_80097F3C = D_80097F4A;
+        break;
+    }
+}
+
+void func_80124164(POLY_GT4* poly, s32 arg1, s32 arg2, s32 arg3, s32 arg4) {
```

Comment:
`arg2` can be renamed as `y`, while `arg3` is a bit more ambiguous... maybe something like `halfHeight` or `radius`? While `arg1` might be something like color intensity